### PR TITLE
SOF 170 - Distance overflow

### PR DESCRIPTION
**Pull request by LiamFly**
Overflow in WP distance and totzulu fixed.

### CODE REVIEW

File: ExtLibs/ArduPilot/CurrentState.cs [hunk 1]
```diff
@@ -849,8 +852,27 @@ public double glide_ratio
         [DisplayText("Dist to WP (dist)")]
         public float wp_dist
         {
-            get => _wpdist * multiplierdist;
-            set => _wpdist = value;
+            get
+            {
+                if (_wpdist < 65535 || lat == 0 && lng == 0 || next_lat == 0)
+                    return _wpdist * multiplierdist;
```
Comment (rob-flylogix) [inline ExtLibs/ArduPilot/CurrentState.cs:858]: Could we add:
if (_wpdist < 65535 || lat == 0 && lng == 0 || next_lat == 0 **&& next_lng == 0)**

Comment (rob-flylogix) [reply ExtLibs/ArduPilot/CurrentState.cs:858]: maybe put in brackets to force precedence i.e.:
if ((_wpdist < 65535) || (lat == 0 && lng == 0) || (next_lat == 0 && next_lng == 0))

**Review by rob-flylogix — CHANGES_REQUESTED** [review]
Could we fix line 857 with:

if ((_wpdist < 65535) || (lat == 0 && lng == 0) || (next_lat == 0 && next_lng == 0))

**Comment by LiamFly** [comment]
> Could we fix line 857 with:
> 
> if ((_wpdist < 65535) || (lat == 0 && lng == 0) || (next_lat == 0 && next_lng == 0))

Done

File: ExtLibs/ArduPilot/CurrentState.cs (hunk 1 above)
Comment (LiamFly) [reply ExtLibs/ArduPilot/CurrentState.cs:858]: Done

**Review by rob-flylogix — APPROVED** [review]

**Pull request merged**